### DISCUSSION TITLE
exportArchive: No "iOS App Store" profiles for team 'TEAMID' matching 'app_name-app-store' are installed.

### DISCUSSION THREAD

**Discussion by iib0011** [codemagic-ci-cd/codemagic-docs#1887] [Q&A]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

React Native

### Steps to reproduce

1.Set up provisioning profile
2.Try to build ios

### Expected results

It should work

### Actual results

It throws this error exportArchive: No "iOS App Store" profiles for team 'TEAM_ID' matching 'app_name-app-store' are installed.

### Build id (optional)

64b03e8e6c4327405f6db867

**Comment by dtrdic** [comment]
Hey @iib0011, could you please check in Xcode if the bundle identifier for your Flutter project matches the bundle identifier of your provisioning profile?

**Reply by dtrdic** [reply]
also i noticed that you added this line inside `Build ipa for distribution` step;
`--archive-xcargs "CODE_SIGNING_REQUIRED=YES CODE_SIGNING_ALLOWED=NO"`
this was needed when Xcode 14.0 was introduced, please remove it and run a build again

**Reply by iib0011** [reply]
Without this line, the Xcode build command fails

**Reply by iib0011** [reply]
> Hey @iib0011, could you please check in Xcode if the bundle identifier for your Flutter project matches the bundle identifier of your provisioning profile?

Yes they match. It's a React native project by the way.

**Reply by dtrdic** [reply]
can you please try remove both archive flags
`--archive-flags "-destination 'generic/platform=iOS'`
`--archive-xcargs "CODE_SIGNING_REQUIRED=YES CODE_SIGNING_ALLOWED=NO"`
and share with me the build ID

**Reply by iib0011** [reply]
Here is the build id: 64b80da30c1bb1d02566546c

**Reply by dtrdic** [reply]
hey @iib0011 i can't see a build where you removed only this piece of code
`--archive-xcargs "CODE_SIGNING_REQUIRED=YES CODE_SIGNING_ALLOWED=NO"`
try running a build with this command
```
xcode-project build-ipa \
  --workspace "$CM_BUILD_DIR/ios/$XCODE_WORKSPACE" \
  --scheme "$XCODE_SCHEME" \
  --archive-flags "-destination 'generic/platform=iOS'"
```

**Reply by iib0011** [reply]
Here is the build id: 64b8f3d517cda3ed416b27fc

**Reply by iib0011** [reply]
@dtrdic can we please setup a video call Tuesday 12pm GMT to fix this issue?

**Reply by dtrdic** [reply]
Hey @iib0011, we provide support via chat. I'll do my best to assist you. Could you please share your yaml file with me? If there's any sensitive information, feel free to hide it.

**Reply by iib0011** [reply]
Here it is
```yaml
workflows:
  ios-workflow:
    name: iOS Workflow
    integrations:
      app_store_connect: app_name-ios-api-key
    environment:
      xcode: 14.2
      ios_signing:
#        distribution_type: app_store
        bundle_identifier: com.app_name.ios
        provisioning_profiles:
          - app_name-app-store-profile
      vars:
        BUNDLE_ID: 'com.app_name.ios'
        XCODE_WORKSPACE: 'App_name.xcworkspace' # <-- Name of your Xcode workspace
        XCODE_SCHEME: 'App_name' # <-- Name of your Xcode scheme
        APP_ID: 1538541364
    scripts:
      - name: Install dependencies
        script: |
          yarn install
      - name: Set Info.plist values
        script: |
          PLIST=$CM_BUILD_DIR/ios/$XCODE_SCHEME/Info.plist
          PLIST_BUDDY=/usr/libexec/PlistBuddy
          $PLIST_BUDDY -c "Add :ITSAppUsesNonExemptEncryption bool false" $PLIST
      - name: Install CocoaPods dependencies
        script: |
          cd ios && pod install
      - name: Set up provisioning profiles settings on Xcode project
        script: |
          sudo gem install xcodeproj
          sudo xcode-project use-profiles --project ios/*.xcodeproj
      - name: Bundle with npx react-native
        script: |
          export NODE_OPTIONS=--openssl-legacy-provider
          npx react-native bundle --entry-file index.js --platform ios --dev false --bundle-output ios/main.jsbundle --assets-dest ios
      - name: Build ipa for distribution
        script: |
          sudo xcode-project build-ipa \
            --workspace "$CM_BUILD_DIR/ios/$XCODE_WORKSPACE" \
            --scheme "$XCODE_SCHEME" \
            --archive-flags "-destination 'generic/platform=iOS'"
    artifacts:
      - build/ios/ipa/*.ipa
      - /tmp/xcodebuild_logs/*.log
      - $HOME/Library/Developer/Xcode/DerivedData/**/Build/**/*.app
      - $HOME/Library/Developer/Xcode/DerivedData/**/Build/**/*.dSYM
    publishing:
      email:
        recipients:
          - ibracool99@gmail.com
        notify:
          success: true
          failure: true
  android-workflow:
    name: Android Workflow
    environment:
      android_signing:
        - app_name-android
      vars:
        PACKAGE_NAME: 'com.app_name'
    scripts:
      - name: Install npm dependencies
        script: |
          yarn install
      - name: Set Android SDK location
        script: |
          echo "sdk.dir=$ANDROID_SDK_ROOT" > "$CM_BUILD_DIR/android/local.properties"
      - name: Bundle Android release
        script: |
          npx react-native bundle --platform android --dev false --entry-file index.js --bundle-output android/app/src/main/assets/index.android.bundle --assets-dest android/app/src/main/res/
          cd android
          chmod +x gradlew
          export NODE_OPTIONS=--openssl-legacy-provider
          ./gradlew bundleRelease
      - name: Assemble Android release
        script: |
          cd android
          chmod +x gradlew
          export NODE_OPTIONS=--openssl-legacy-provider
          ./gradlew assembleRelease

    artifacts:
      - android/app/build/outputs/**/*.aab
      - android/app/build/outputs/**/*.apk
    publishing:
      email:
        recipients:
          - ibracool99@gmail.com
        notify:
          success: true
          failure: true

**Reply by dtrdic** [reply]
Please uncomment `distribution_type: app_store` as it is used to fetch all uploaded signing files matching the app_store distribution type and bundle identifier during the build.

Additionally, please note that when using the fields `distribution_type` and `bundle_identifier`, it is not allowed to configure the `provisioning_profiles` field. So, please comment out this field as well:
```
 provisioning_profiles:
          - app_name-app-store-profile
```

**Reply by iib0011** [reply]
I originally had the `distribution_type` set, without the `provisioning_profiles`. It didn't work, that is why I changed it.

**Reply by iib0011** [reply]
I made the changes you asked. Here is the build id: 64be5f356589b98b3388e42f

**Reply by dtrdic** [reply]
Okay, have you set up the iOS bundle identifier inside the app.json file, as mentioned [here](https://docs.codemagic.io/yaml-quick-start/building-a-react-native-app/#setting-up-the-android-package-name-and-ios-bundle-identifier)? 
Also, let's follow along with this [sample project](https://github.com/codemagic-ci-cd/codemagic-sample-projects/blob/main/react-native/react-native-demo-project/codemagic.yaml#L50) for reference. Please modify it in the same way.
And please share a screenshot of your code project
e.g
<img width="542" alt="image" src="https://github.com/codemagic-ci-cd/codemagic-docs/assets/38918425/c795bd35-fdde-41fe-a16f-037d3714528d">

**Reply by iib0011** [reply]
We don't use expo, so we didn't add the bundle identifier in the app.json file. I used the sample project for reference.

**Reply by dtrdic** [reply]
sorry i meant a screenshot of your xcode project

**Reply by iib0011** [reply]
Here it is![IMG-20230725-WA0005.jpg](https://github.com/codemagic-ci-cd/codemagic-docs/assets/58892076/c742ea52-2070-443c-a5eb-d23bf2d90441)

**Reply by dtrdic** [reply]
Can you try switching the positions of steps 7 and 8? First, run `npx react-native bundle`, and then use `xcode-project use-profiles`.

**Reply by iib0011** [reply]
Still not working. Here is the build id: 64bf92d87c4a904d9360d6cd

**Reply by dtrdic** [reply]
can you please add a `--verbose` flag on the step that is failing? the error message is quite generic, so let's see what more information can we get from it

**Reply by iib0011** [reply]
Here is the build id: 64bfb21cf2a1d76b38799039

**Reply by dtrdic** [reply]
can you tell me a little bit about your local configuration? xcode version, node version, what type of machine you are using m1/mac pro/m2? also do you face this issue locally as well?

**Reply by dtrdic** [reply]
i created one react native app as well, and used a part of your script, so please try with this one.. because i can't see a build where you switched steps 7 and 8, as I mentioned above
```
 scripts:
                - name: Install npm dependencies
                  script: |
                    npm install
                - name: Set Info.plist values
                  script: |
                    # This allows publishing without manually answering the question about encryption 
                    PLIST=$CM_BUILD_DIR/$XCODE_SCHEME/Info.plist
                    PLIST_BUDDY=/usr/libexec/PlistBuddy
                    $PLIST_BUDDY -c "Add :ITSAppUsesNonExemptEncryption bool false" $PLIST
                - name: Install CocoaPods dependencies
                  script: |
                    cd ios && pod install
                - name: Set up code signing settings on Xcode project
                  script: |
                    xcode-project use-profiles 
                - name: Bundle with npx react-native
                  script: |
                    export NODE_OPTIONS=--openssl-legacy-provider
                    npx react-native bundle --entry-file index.js --platform ios --dev false --bundle-output ios/main.jsbundle --assets-dest ios
                - name: Increment build number
                  script: |
                    cd $CM_BUILD_DIR/ios
                    LATEST_BUILD_NUMBER=$(app-store-connect get-latest-app-store-build-number "$APP_ID")
                    agvtool new-version -all $(($LATEST_BUILD_NUMBER + 1))
                - name: Build ipa for distribution
                  script: |
                    xcode-project build-ipa \
                      --workspace "$CM_BUILD_DIR/ios/$XCODE_WORKSPACE" \
                      --scheme "$XCODE_SCHEME"
```

**Reply by iib0011** [reply]
In This build I switched the 2 steps: 64bf92d87c4a904d9360d6cd

**Reply by dtrdic** [reply]
Just try the script that i shared, maybe we didn't understand each other.
`Set up provisioning profiles settings on Xcode project`(or in my case that I shared with you **Set up code signing settings on Xcode project**) must come before `Bundle with npx react-native` step

**Reply by iib0011** [reply]
Here is the build id: 64c1093c2745cdac99238ff0

**Reply by dtrdic** [reply]
please try without `--project ios/*.xcodeproj` flag on the `Set up provisioning profiles settings on Xcode project` step.
if that doesn't help try with my script that I shared above

**Reply by iib0011** [reply]
Thank you so much for your help, but it still doesn't work. I finally give up.
